sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -741,7 +741,7 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
 
   def variableSubstituteDepth: Int = getConf(VARIABLE_SUBSTITUTE_DEPTH)
 
-  def warehousePath: String = new Path(getConf(WAREHOUSE_PATH)).toString
+  def warehousePath: String = Utils.resolveURI(getConf(WAREHOUSE_PATH)).toString
```

Comment:
So, I know I suggested this, but looking at it now, it seems wrong.

Before, if a user explicitly set this value in the configuration to `"/user/blah/warehouse"`, that would be interpreted as an HDFS path (or rather a path under `fs.defaultFS`). With this change it would be a local path. Right? So `Utils.resolveURI` should really only apply to the default value.